# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: docshell/base/CanonicalBrowsingContext.h
# Commit: 77d15266af3c
# Full Hash: 77d15266af3c0a014051fb57f50356e4acd9e880
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2020-06-18 04:43:29
# Regressor Bug: 1646088
# File Overlap Count: 2
# Description:
#   Bug 1646088 - Part 1: Keep processes alive during process switches, r=kmag
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D79888
# ==============================================================================

diff -r fa227b47dac7 -r 77d15266af3c docshell/base/CanonicalBrowsingContext.h
--- a/docshell/base/CanonicalBrowsingContext.h	Wed Jun 17 09:41:24 2020 +0000
+++ b/docshell/base/CanonicalBrowsingContext.h	Wed Jun 17 15:43:35 2020 +0000
@@ -205,13 +205,14 @@
     friend class CanonicalBrowsingContext;
 
     ~PendingRemotenessChange();
-    void ProcessReady(ContentParent* aContentParent);
-    void Finish(ContentParent* aContentParent);
+    void ProcessReady();
+    void Finish();
     void Clear();
 
     RefPtr<CanonicalBrowsingContext> mTarget;
     RefPtr<RemotenessPromise::Private> mPromise;
     RefPtr<GenericPromise> mPrepareToChangePromise;
+    RefPtr<ContentParent> mContentParent;
 
     uint64_t mPendingSwitchId;
     bool mReplaceBrowsingContext;